src/mds/Server.cc

```diff
@@ -5748,9 +5748,12 @@ int Server::parse_quota_vxattr(string name, string value, quota_info_t *quota)
           return r;
       }
     } else if (name == "quota.max_bytes") {
-      int64_t q = boost::lexical_cast<int64_t>(value);
-      if (q < 0)
+      string cast_err;
+      int64_t q = strict_iec_cast<int64_t>(value, &cast_err);
+      if(!cast_err.empty()) {
+        derr << cast_err << dendl;
```

Comment:
So, when a negative number is encountered, it will hit this line https://github.com/ceph/ceph/blob/main/src/common/strtol.cc#L190 and the err string `cast_err` in `Server.cc` would be non-empty and thus will print the msg there at line 5754 and return `-CEPHFS_EINVAL` and so `quota->max_bytes = q` would not happen

Reply:
@dparmar18 Include this in the tests.

Reply:
@vshankar do you mean testing negative numbers? If so then it has already been taken care of, test case included in the PR - https://github.com/ceph/ceph/pull/49008/commits/164a1726cf4ac0657d56018dbe5f30121368a15b covers every possible case :)

Reply:
Thx!